src/ray/core_worker/lib/java/jni_utils.h

```diff
@@ -344,4 +345,24 @@ inline jobject NativeRayObjectToJavaNativeRayObject(
   return java_obj;
 }
 
+inline jobject NativeRayFunctionDescriptorToJavaStringList(
+    JNIEnv *env, const ray::FunctionDescriptor &function_descriptor) {
+  if (function_descriptor->Type() ==
+      ray::FunctionDescriptorType::kJavaFunctionDescriptor) {
+    auto typed_descriptor = function_descriptor->As<ray::JavaFunctionDescriptor>();
+    std::vector<std::string> function_descriptor_list = {typed_descriptor->ClassName(),
+                                                         typed_descriptor->FunctionName(),
+                                                         typed_descriptor->Signature()};
+    return NativeStringVectorToJavaStringList(env, function_descriptor_list);
+  } else if (function_descriptor->Type() ==
+             ray::FunctionDescriptorType::kPythonFunctionDescriptor) {
+    auto typed_descriptor = function_descriptor->As<ray::PythonFunctionDescriptor>();
+    std::vector<std::string> function_descriptor_list = {
+        typed_descriptor->ModuleName(), typed_descriptor->ClassName(),
+        typed_descriptor->FunctionName(), typed_descriptor->FunctionHash()};
+    return NativeStringVectorToJavaStringList(env, function_descriptor_list);
+  }
+  return NativeStringVectorToJavaStringList(env, std::vector<std::string>());
```

Comment:
How about `RAY_LOG(FATAL)` here?

Reply:
I will do it if Java worker can assert fatal.

Reply:
Only driver task spec have function descriptor which Type() == ray::FunctionDescriptorType::FUNCTION_DESCRIPTOR_NOT_SET, So RAY_LOG(FATAL) here is OK.